### PR TITLE
Lambda & Stream APIs

### PR DESCRIPTION
**Pull request by southernwestern**
https://www.notion.so/Lambda-Stream-APIs-1897c27153f94308908a2a2e7f7bafcb

### CODE REVIEW

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      javaGrepLambdaImp.process();
+    } catch (Exception ex) {
+      ex.printStackTrace();
+    }
+
+  }
+
+  @Override
+  public void process() throws IOException {
+    File rootDir = new File(this.getOutFile());
```
Comment (zhenzhangca) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: where is getOutFile method in this class?

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    List<String> fileData = readLines(rootDir);
+
+    rootDir.delete();
+    rootDir.createNewFile();
+
+    fileData.addAll(listFiles(this.rootPath)
```
Comment (zhenzhangca) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: where is rootPath field in this class?

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            .stream()
+            .map(File::getPath)
+            .filter(line -> !fileData.contains(line) &&  containsPattern(line))
```
Comment (zhenzhangca) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: where is containsPattern method in this class?

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java (hunk 1 above)
Comment (southernwestern) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: The class JavaGrepLambdaImp inherits all the mentioned methods from JavaGrepImp.

**Pull request merged**